hugr-core/src/hugr/serialize/test.rs

```diff
@@ -517,6 +517,18 @@ fn roundtrip_optype(#[case] optype: impl Into<OpType> + std::fmt::Debug) {
     });
 }
 
+#[test]
+// test all standard extension serialisations are valid against scheme
+fn std_extensions_valid() {
+    let std_reg = crate::std_extensions::std_reg();
+    for (_, ext) in std_reg.into_iter() {
+        let val = serde_json::to_value(ext).unwrap();
+        NamedSchema::check_schemas(&val, get_schemas(true));
+        // check deserialises correctly, can't check equality because of custom binaries.
```

Comment:
However I think you could try reserializing the deserialized Extension to check equality with the original?
```
deserialize(serialize(ext)) != ext // because custom binaries
serialize(deserialize(serialize(ext))) == serialize(ext) // I think

Reply:
yes nice thanks